nihms-filesystem-builder/src/main/java/org/dataconservancy/nihms/builder/fs/FilesystemModelBuilder.java

```diff
@@ -18,17 +18,164 @@
  */
 public class FilesystemModelBuilder implements SubmissionBuilder {
 
     @Override
     public NihmsSubmission build(String formDataUrl) throws InvalidModel {
-        return null;
+        Properties properties = new Properties();
+        InputStream is = null;
+
+        //The submission object to populate
+        NihmsSubmission submission = new NihmsSubmission();
+
+        //The file to add, and its list to add to submission
+        NihmsFile file = new NihmsFile();
+        List<NihmsFile> files = new ArrayList<>();
+
+        //The metadata object, and components to add to it ...
+        NihmsMetadata metadata = new NihmsMetadata();
+        NihmsMetadata.Journal journal = new NihmsMetadata.Journal();
+        NihmsMetadata.Manuscript manuscript = new NihmsMetadata.Manuscript();
+
+        //... including the person object and its list
+        NihmsMetadata.Person person = new NihmsMetadata.Person();
+        List<NihmsMetadata.Person> persons = new ArrayList<>();
+
+        try {
+            is = new FileInputStream(formDataUrl);
+            if (is == null) {
+                System.out.println("Sorry, unable to find submission properties file " + formDataUrl);
+                return null;
+            }
+
+            properties.load(is);
+
+            Enumeration<?> e = properties.propertyNames();
+            while (e.hasMoreElements()) {
+                String key = (String) e.nextElement();
+                String value = properties.getProperty(key);
+
+                switch (key) {
+                    //the id for the submission
+                    case NihmsBuilderPropertyNames.NIHMS_SUBMISSION_ID:
+                        submission.setId(value);
+                        break;
+
+                    //file properties
+                    case NihmsBuilderPropertyNames.NIHMS_FILE_NAME:
+                        file.setName(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_FILE_LABEL:
+                        file.setLabel(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_FILE_LOCATION:
+                        file.setLocation(value);
+                        break;
+
+                    //journal metadata
+                    case NihmsBuilderPropertyNames.NIHMS_JOURNAL_ID:
+                        journal.setJournalId(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_JOURNAL_ISSN:
+                        journal.setIssn(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_JOURNAL_TITLE:
+                        journal.setJournalTitle(value);
+                        break;
+
+                    //manuscript metadata
+                    case NihmsBuilderPropertyNames.NIHMS_MANUSCRIPT_DOI:
+                        try {
+                            manuscript.setDoi(new URI(value));
```

Comment:
Another option here is to use `URI.create(...)`, and then you don't have to mess with exception handling.  Since we aren't being super picky with this implementation, I think it's totally ok to allow `URI.create(...)` to throw a runtime exception if something is malformed in the properties.